### PR TITLE
#2976. Add more type inference tests

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

**Comment by sgrekhov** [comment]
Ready for review now.

File: LanguageFeatures/Static-access-shorthand/type_inference_A02_t05.dart [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+// [cfe] unspecified
+  } else {
+    return const .id();
+//               ^
+// [analyzer] unspecified
+// [cfe] unspecified
+  }
+}
+
+(int,) getRecord() {
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Static-access-shorthand/type_inference_A02_t05.dart]: Added

File: LanguageFeatures/Static-access-shorthand/type_inference_A02_t05.dart [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+// [cfe] unspecified
+  } else {
+    return const .id();
+//               ^
+// [analyzer] unspecified
+// [cfe] unspecified
+  }
+}
+
+Function getFunction() {
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Static-access-shorthand/type_inference_A02_t05.dart]: Added

File: LanguageFeatures/Static-access-shorthand/type_inference_A03_t01.dart [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+// [cfe] unspecified
+  } else {
+    return .id;
+//         ^
+// [analyzer] unspecified
+// [cfe] unspecified
+  }
+}
+
+(int,) getRecord() {
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Static-access-shorthand/type_inference_A03_t01.dart]: Added

File: LanguageFeatures/Static-access-shorthand/type_inference_A03_t01.dart [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+// [cfe] unspecified
+  } else {
+    return .id;
+//         ^
+// [analyzer] unspecified
+// [cfe] unspecified
+  }
+}
+
+Function getFunction() {
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Static-access-shorthand/type_inference_A03_t01.dart]: And here

File: LanguageFeatures/Static-access-shorthand/type_inference_A04_t01.dart [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  e0;
+  E get id => E.e0;
+}
+
+extension type ET.foo(int _) {
+  ET get id => this;
+}
+
+main() {
+  C c1 = .new();
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Static-access-shorthand/type_inference_A04_t01.dart]: Surprisingly, we don't have a specific case where we attempt to invoke an unnamed constructor on a class or extension type that doesn't have one. Moved this case (using `.new()`) to `type_inference_A05_t02.dart` with an appropriate description. Removed references to `.new()` from this test description.

**Review by sgrekhov — COMMENTED** [review]
Updated. PTAL.

**Review by eernstg — APPROVED** [review]
LGTM

File: LanguageFeatures/Static-access-shorthand/type_inference_A04_t01.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (eernstg) [reply LanguageFeatures/Static-access-shorthand/type_inference_A04_t01.dart]: SG, thanks!

**Pull request merged**